app/models/recipe_ingredient.rb

```diff
@@ -1,3 +1,6 @@
 class RecipeIngredient < ApplicationRecord
-
+  validates :recipe_id, presence: true
```

Comment:
You don't need to add validations to the join table. This could be where you got a little snagged up with `:id`.